### PR TITLE
PowerShell: parse a colon following the class name

### LINKED ISSUE #3530: PowerShell: When inheriting a class, the output is not expected when there is no space between the class name and the ':'.
The content of input file:

```powershell
class Derived : Base {
}

class Derived2: Base {
}
```

The tags output you are not satisfied with:

```
Derived   input.ps1  /^class Derived : Base {$/;"    c
Derived2: input.ps1  /^class Derived2: Base {$/;"    c
...
```

The tags output you expect:

```
Derived   input.ps1  /^class Derived : Base {$/;"    c
Derived2  input.ps1  /^class Derived2: Base {$/;"    c
...
```

The version of ctags:

```
$ ctags --version
Universal Ctags 5.9.0(83c1a093), Copyright (C) 2015-2022 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Nov 13 2022, 16:57:39
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +gnulib_regex, +iconv, +option-directory, +packcc, +optscript
```

How do you get ctags binary: [daily builds](https://github.com/universal-ctags/ctags-win32/releases)

### PR DESCRIPTION
**Pull request by kumarstack55**
closes #3530

### CODE REVIEW

File: parsers/powershell.c [hunk 1]
```diff
@@ -236,6 +241,26 @@ static void parseString (vString *const string, const int delimiter)
...
+ * 	FunctionName
+ * 	local:VariableName
+ * 	private:FunctionName
+ */
+static void parseScopeIdentifier (vString *const string, const int firstChar)
+{
+	int c = firstChar;
+	do
+	{
+		vStringPut (string, (char) c);
```
Comment (masatake) [inline parsers/powershell.c:257]: The cast operation, (char), is not needed. vStringPut takes an integer as the second arg.

Comment (masatake) [reply parsers/powershell.c]: I think a commit should be perfect and should not depend on a successor commit (if possible).

[283e2fd](https://github.com/universal-ctags/ctags/pull/3531/commits/283e2fd0710119bd5e0f38f91e80e695340f4570) is not perfect; it includes an unnecessary cast operation. It is fixed in the successor commit ([6c60f00](https://github.com/universal-ctags/ctags/pull/3531/commits/6c60f004c9ed79b148ceca8a3212fd756cc76a90)). 

If a new potential contributor reads [283e2fd](https://github.com/universal-ctags/ctags/pull/3531/commits/283e2fd0710119bd5e0f38f91e80e695340f4570) to study how to improve the Powershell parser, the one may think (char) should be added to the second arg of vStringPut.

I think you should reorder the commits in this pull request.

Make a commit [6c60f00](https://github.com/universal-ctags/ctags/pull/3531/commits/6c60f004c9ed79b148ceca8a3212fd756cc76a90) as a preparation change, then add the main commit to this pull request.

If you don't know how to do this, it is time to learn "git rebase -i ...".
I think it is the most brilliant feature git provides.

**Review by masatake — APPROVED** [review]

**Comment by masatake** [comment]
Thank you.

**Comment by masatake** [comment]
After merging your code, I found your change looked tricky.

In readToken of powershell.c you did:
```c
			else
			{
				if (token->keyword == KEYWORD_function ||
						token->keyword == KEYWORD_filter)
					parseScopeIdentifier (token->string, c);
				else
					parseIdentifier (token->string, c);
```

You refer to `token->keyword` here. This a bad idea; readTags takes `token` as an argument for output. For input, we should introduce a new parameter.

I will make a pull request.

**Comment by masatake** [comment]
I opened #3533.

**Comment by kumarstack55** [comment]
Thanks for the merge.

I seem to have made a mistake in how I implemented it.
As you said, it turned out to be a tricky implementation.

Thank you for correcting it.

**Pull request merged**